contracts/token/ERC1155/ERC1155.sol

```diff
@@ -183,12 +188,35 @@ abstract contract ERC1155 is Context, ERC165, IERC1155, IERC1155MetadataURI, IER
             uint256 id = ids.unsafeMemoryAccess(0);
             uint256 amount = amounts.unsafeMemoryAccess(0);
             emit TransferSingle(operator, from, to, id, amount);
-            if (to != address(0)) {
-                _doSafeTransferAcceptanceCheck(operator, from, to, id, amount, data);
-            }
         } else {
             emit TransferBatch(operator, from, to, ids, amounts);
-            if (to != address(0)) {
+        }
+    }
+
+    /**
+     * @dev Version of {_update} that performs the token acceptance check by calling {IERC1155Receiver-onERC1155Received}
+     * or {IERC1155Receiver-onERC1155BatchReceived} on the receiver address if it contains code (eg. is a smart contract
+     * at the moment of execution).
+     *
+     * IMPORTANT: Overriding this function is discouraged because it poses a reentrancy risk from the receiver. So any
+     * update to the contract state after this function would break the check-effect-interaction pattern. Consider
+     * overriding {_update} instead.
+     */
+    function _updateChecked(
```

Comment:
`_updateWithAcceptanceCheck` ?

Reply:
I find `_updateChecked` better than `_updateWithAcceptanceCheck`, but I'd suggest `_acceptedUpdate`

Reply:
Out of the two-word options I prefer `_updateChecked`. 

`_updateWithAcceptanceCheck` is good too but I agree shorter is better.

Reply:
I'd value clarity over shortness (it's not that long when compared to "_doBatchTransferAcceptanceCheck"). But either is fine by me.

If you both prefer "_updateChecked" let's just do that

Reply:
Is there a risk that "Checked" is interpreted by some devs as the opposite of "unchecked {}" ?

Reply:
Fair point! I think there's a reasonable risk and I'd favor clarity in that case. Still, `_updateWithAcceptanceCheck` doesn't fully convince me, what about `_updateAcceptanceChecked`?

Reply:
Ok after reconsidering I think `_updateWithAcceptanceCheck` didn't convince me because there's a connector. But it's the same case for `_upgradeAndCall`, so let's go with `_updateWithAcceptanceCheck`.